contracts/test/unit/oracle/VoltSystemOracle.t.sol

```diff
@@ -0,0 +1,289 @@
+            voltSystemOracle.compoundInterest(); /// compound interest periods amount of times
+            expectedOraclePrice = _calculateDelta(
+                expectedOraclePrice,
+                voltSystemOracle.annualChangeRateBasisPoints() +
+                    Constants.BASIS_POINTS_GRANULARITY
+            );
+            assertEq(expectedOraclePrice, voltSystemOracle.oraclePrice());
+        }
+    }
+
+    function testSetup() public {
+        assertEq(voltSystemOracle.oraclePrice(), startPrice);
+        assertEq(
+            voltSystemOracle.annualChangeRateBasisPoints(),
+            annualChangeRateBasisPoints
+        );
+        assertEq(voltSystemOracle.periodStartTime(), startTime);
+        assertEq(voltSystemOracle.getCurrentOraclePrice(), startPrice);
+    }
+
+    function testCompoundBeforeInitFails() public {
+        vm.expectRevert("VoltSystemOracle: not past end time");
+        voltSystemOracle.compoundInterest();
+    }
+
+    function testCompoundSucceedsAfterOneYear() public {
+        assertEq(voltSystemOracle.oraclePrice(), startPrice);
+        vm.warp(block.timestamp + voltSystemOracle.periodStartTime());
+
+        uint256 oraclePrice = voltSystemOracle.oraclePrice();
+
+        vm.warp(block.timestamp + 365 days);
+
+        uint256 expectedOraclePrice = _calculateDelta(
+            oraclePrice,
+            annualChangeRateBasisPoints + Constants.BASIS_POINTS_GRANULARITY
+        );
+        assertEq(voltSystemOracle.getCurrentOraclePrice(), expectedOraclePrice);
+        voltSystemOracle.compoundInterest();
+        assertEq(voltSystemOracle.oraclePrice(), expectedOraclePrice);
+    }
+
+    function testLinearInterpolation() public {
+        assertEq(voltSystemOracle.oraclePrice(), startPrice);
+        vm.warp(voltSystemOracle.periodStartTime() + 365 days);
+
+        assertEq(
+            voltSystemOracle.getCurrentOraclePrice(),
+            _calculateDelta(
+                startPrice,
+                annualChangeRateBasisPoints + Constants.BASIS_POINTS_GRANULARITY
+            )
+        );
+    }
+
+    /// assert that price cannot increase before block.timestamp of 100,000
+    /// since uint16 max is 65,535, the current oracle price cannot increase
+    function testNoLinearInterpolationBeforeStartTime(uint16 x) public {
+        assertEq(voltSystemOracle.oraclePrice(), startPrice);
+        vm.warp(block.timestamp + x);
+
+        assertEq(
+            voltSystemOracle.getCurrentOraclePrice(),
+            voltSystemOracle.oraclePrice()
+        );
+    }
+
+    function testLinearInterpolationOnlyAfterStartTime(uint16 x) public {
+        assertEq(voltSystemOracle.oraclePrice(), startPrice);
+        vm.warp(block.timestamp + x);
+        uint256 cachedOraclePrice = voltSystemOracle.oraclePrice();
+
+        if (x > voltSystemOracle.periodStartTime()) {
+            /// interpolation because past start time
+            uint256 timeDelta = block.timestamp -
```

Comment:
express time delta as a decimal, then use that to calculate the amount of basis points the price should change, then apply that change to the price to prove that the formula is correct.